apps/backend/src/controllers/agri/bpp/search.ts

```diff
@@ -0,0 +1,76 @@
+import { NextFunction, Request, Response } from "express";
+import fs from "fs";
+import path from "path";
+import YAML from "yaml";
+import {
+	responseBuilder,
+	AGRI_EXAMPLES_PATH,
+	logger,
+} from "../../../lib/utils";
+import { ON_ACTION_KEY } from "../../../lib/utils/actionOnActionKeys";
+import { SERVICES_DOMAINS } from "../../../lib/utils/apiConstants";
+
+export const searchController = (
+	req: Request,
+	res: Response,
+	next: NextFunction
+) => {
+	try {
+		const domain = req?.body?.context?.domain;
+		const {scenario}=req.query
```

Comment:
This format of taking the scenario from request query is wrong. No NP is supposed to send in the scenario like this. Please correct this after a discussion with @tanya-ondc

Reply:
*correct across all instances of such req.query usage.